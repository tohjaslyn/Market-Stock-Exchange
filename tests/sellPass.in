add trader1 100
import trader1 ABC 10
add trader2 100
buy trader2 ABC 10 6
sell trader1 ABC 10 5
import trader1 ABC 10
buy trader2 ABC 3 6
buy trader2 ABC 3 5.50
sell trader1 ABC 10 5
import trader1 DEF 20
buy trader2 DEF 7 6
buy trader2 DEF 3 6.50
sell trader1 DEF 10 5
sell trader1 DEF 10 5
import trader1 GHI 20
buy trader2 GHI 7 6
buy trader2 GHI 3 6.50
sell trader1 GHI 9 5
exit
