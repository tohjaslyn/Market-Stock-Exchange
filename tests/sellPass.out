$ Success.
$ Trader now has 10.00 units of ABC.
$ Success.
$ No trades could be made, order added to buy book.
$ Product sold in entirety, trades as follows:
trader1->trader2: 10.00xABC for $6.00.
$ Trader now has 10.00 units of ABC.
$ No trades could be made, order added to buy book.
$ No trades could be made, order added to buy book.
$ Product sold in part, trades as follows:
trader1->trader2: 3.00xABC for $6.00.
trader1->trader2: 3.00xABC for $5.50.
$ Trader now has 20.00 units of DEF.
$ No trades could be made, order added to buy book.
$ No trades could be made, order added to buy book.
$ Product sold in entirety, trades as follows:
trader1->trader2: 3.00xDEF for $6.50.
trader1->trader2: 7.00xDEF for $6.00.
$ No trades could be made, order added to sell book.
$ Trader now has 20.00 units of GHI.
$ No trades could be made, order added to buy book.
$ No trades could be made, order added to buy book.
$ Product sold in entirety, trades as follows:
trader1->trader2: 3.00xGHI for $6.50.
trader1->trader2: 6.00xGHI for $6.00.
$ Have a nice day.
